keps/0007-dynamic-audit-control.md

```diff
@@ -0,0 +1,126 @@
+---
+kep-number: 7
+title: Dynamic Audit Control
+authors:
+  - "@pbarker"
+owning-sig: sig-auth
+participating-sigs:
+  - sig-cluster-lifecycle
+reviewers:
+  - TBD
+approvers:
+  - TBD
+editor: TBD
+creation-date: 2018-05-18
+last-updated: 2018-05-18
+status: provisional
+see-also:
+  - need to find dynamic admission KEP
+---
+
+# Dynamic Audit Control
+
+## Table of Contents
+
+* [Dynamic Audit Control](#dynamic-audit-control)
+  * [Table of Contents](#table-of-contents)
+  * [Summary](#summary)
+  * [Motivation](#motivation)
+      * [Goals](#goals)
+      * [Non-Goals](#non-goals)
+  * [Proposal](#proposal)
+      * [Dynamic Configuration of Policy](#dynamic-configuration-of-policy)
+      * [Dynamic Configuration of Flags](#dynamic-configuration-of-flags)
+      * [User Stories](#user-stories)
+        * [Story 1](#story-1)
+        * [Story 2](#story-2)
+      * [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+      * [Risks and Mitigations](#risks-and-mitigations)
+  * [Graduation Criteria](#graduation-criteria)
+  * [Implementation History](#implementation-history)
+  * [Alternatives](#alternatives)
+
+## Summary
+
+We want to allow the advanced auditing features to be dynamically configured. Following in the same vein as [Dynamic Admission Control](https://kubernetes.io/docs/admin/extensible-admission-controllers/) we would like to provide a means of configuring the auditing features post cluster provisioning.
+
+## Motivation
+
+The advanced auditing features are a powerfull tool, yet difficult to configure. The configuration requires deep insight into the deployment mechanism of choice and often takes many iterations to configure properly requiring a restart of the apiserver each time. Moreover, the ability to install addon tools that configure and enhance audting is hindered by the overhead in configuration. Such tools frequently run on the cluster requiring future knowledge of how to reach them when the cluster is live. These tools could enhance the security and conformance of the cluster and its applications.
+
+### Goals
+- Provide an api and set of objects to configure the advanced auditing kube-apiserver webhook flags dynamically
+- Allow the current audit policy object to be configured dynamically
+
+### Non-Goals
+- Provide a generic interface to configure all kube-apiserver flags
+
+## Proposal
+
+### Dynamic Configuration of Policy
```

Comment:
It's useful to log the changes to the `Policy` in the audit log, then it's easy to tell if changes in the audit log is a result of policy changes. To enforce this, we can hard code the rule for `create/update/delete/patch` of `audit.k8s.io/v1beta1/policy` to always be `level=RequestResponse`, and is not subject to the global `omitStages`.

Reply:
good idea 👍 will add